internal/registry/acctest/par.go

```diff
@@ -0,0 +1,310 @@
+package acctest
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	structenv "github.com/caarlos0/env/v6"
+	packer_service "github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/preview/2021-04-30/client/packer_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/preview/2021-04-30/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/packer/acctest"
+	"github.com/hashicorp/packer/internal/registry"
+	"google.golang.org/grpc/codes"
+)
+
+type hcpConf struct {
+	OrgId        string `env:"HCP_ORG_ID"`
+	ProjectId    string `env:"HCP_PROJECT_ID"`
+	ApiHost      string `env:"HCP_API_HOST"`
+	AuthUrl      string `env:"HCP_AUTH_URL"`
+	ClientId     string `env:"HCP_CLIENT_ID"`
+	ClientSecret string `env:"HCP_CLIENT_SECRET"`
+	UserAgent    string `env:"HCP_CLIENT_USER_AGENT" envDefault:"packer-par-acc-test"`
+}
+
+type Config struct {
+	*registry.Client
+	Loc *sharedmodels.HashicorpCloudLocationLocation
+	T   *testing.T
+}
+
+func NewTestConfig(t *testing.T) (*Config, error) {
+	checkEnvVars(t)
+	cfg := hcpConf{}
+	if err := structenv.Parse(&cfg); err != nil {
+		t.Errorf("%+v\n", err)
+	}
+	cli, err := registry.NewClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Config{
+		Client: cli,
+		Loc: &sharedmodels.HashicorpCloudLocationLocation{
+			OrganizationID: cfg.OrgId,
+			ProjectID:      cfg.ProjectId,
+		},
+		T: t,
+	}, nil
+}
+
+func checkEnvVars(t *testing.T) {
+	t.Helper()
+	if os.Getenv(acctest.TestEnvVar) == "" {
+		t.Skip(fmt.Sprintf(
+			"Acceptance tests skipped unless env '%s' set",
+			acctest.TestEnvVar))
+		return
+	}
+	if os.Getenv("HCP_CLIENT_ID") == "" {
+		t.Fatal("HCP_CLIENT_ID must be set for acceptance tests")
+	}
+	if os.Getenv("HCP_CLIENT_SECRET") == "" {
+		t.Fatal("HCP_CLIENT_SECRET must be set for acceptance tests")
+	}
+}
+
+// GetIterationByID is a helper to validate that the id for a given bucketSlug returns a valid Iteration ID.
+func (cfg *Config) GetIterationByID(
```

Comment:
nit: In the internal registry pkg you have the methods `GetIteration_byID` and `GetIteration_byFingerprint`, which seem a bit inconsistent with the rest of the code base. Is there any reason why they can't be `GetIerationByID` and `GetIterationByFIngerprint`?

Reply:
Woops, supergood catch, these are doing the same thing !

Reply:
Ehm, good question here, so. Other functions like `DeleteBucket` will do just they say they do.

But, there are multiple ways to get an iteration, either by ID or by Fingerprint, but, there's no way to get an iteration without that option. Actually, I think I will just make it an option parameter.